src/Mvc/Mvc.Core/src/Routing/IRequestDelegateFactory.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    /// <summary>
+    /// Internal interfaces that allows us to optimize the request execution path based on ActionDescriptor
+    /// </summary>
+    internal interface IRequestDelegateFactory
```

Comment:
It is a little odd to merge an abstraction without any implementations. I know there's https://github.com/dotnet/aspnetcore/blob/e9d16525cf3a832631a4e29ce826290adae873a9/src/Mvc/Mvc.Core/src/Routing/ControllerRequestDelegateFactory.cs ofc. Maybe we should wait for that or some other implementation of IRequestDelegateFactory to be merged before taking this.

Reply:
Agreed. I'll merge both PRs